src/main/java/com/chikli/hudson/plugin/naginator/NaginatorRetryAction.java

```diff
@@ -47,8 +56,14 @@ public String getUrlName() {
             "retry" : null;
     }
 
-    public void doIndex(StaplerResponse res, @AncestorInPath AbstractBuild build) throws IOException {
-        build.getACL().checkPermission(Item.BUILD);
+    public void doIndex(StaplerResponse res, @CheckForNull @AncestorInPath AbstractBuild<?, ?> build) throws IOException {
+        if (build == null) {
+            // This should not happen as
+            // this page is displayed only for AbstractBuild.
+            res.sendError(HttpServletResponse.SC_NOT_FOUND);
+            return;
+        }
+        build.getParent().checkPermission(Item.BUILD);
```

Comment:
Now use `AbstractProject#checkPermission()` instead of `AbstractBuild#checkPermission()`.
Actually they are same for almost all cases (might be for all cases for now), but can be different logically.
What we have to check is whether the project is buildable.